### PR TITLE
deno: 1.46.2 -> 1.46.3

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Move to by-name and update. Working on fixing x86 darwin atm but figured I'd do this bump real quick. 

Leaving rusty_v8 on same revision, doesn't build with newest bump.

EDIT: Rust toolchain doesn't currently support overriding the darwin SDK. Will revisit enabling x86 darwin after the darwin SDK refactor PR lands. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
@ofborg eval

**Comment by khaneliman** [comment]
@ofborg eval

File: pkgs/by-name/de/deno/package.nix [hunk 1]
```diff
@@ -102,6 +105,12 @@ rustPlatform.buildRustPackage rec {
...
     mainProgram = "deno";
     maintainers = with maintainers; [ jk ];
-    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    broken = stdenv.isDarwin && stdenv.isx86_64;
```
Comment (GaetanLepage) [inline pkgs/by-name/de/deno/package.nix:114]: Consider adding a comment to give context.

File: pkgs/by-name/de/deno/package.nix [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
 rustPlatform.buildRustPackage rec {
   pname = "deno";
-  version = "1.46.2";
+  version = "1.46.3";
 
   src = fetchFromGitHub {
     owner = "denoland";
-    repo = pname;
+    repo = "deno";
     rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/de/deno/package.nix:21]:
```suggestion
    rev = "refs/tags/v${version}";
```
while we're at it

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Review by 06kellyjac — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 343868`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
    <li>windmill</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
    <li>windmill</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 13 packages marked as broken and skipped:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>era</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>silverbullet</li>
    <li>vimPlugins.ddc-filter-matcher_head</li>
    <li>vimPlugins.ddc-filter-sorter_rank</li>
    <li>vimPlugins.ddc-source-lsp</li>
    <li>vimPlugins.ddc-ui-native</li>
    <li>vimPlugins.ddc-ui-pum</li>
    <li>vimPlugins.ddc-vim</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
  </ul>
</details>

**Pull request merged**